sdk/plugin/grpc_backend_server.go

```diff
@@ -77,6 +77,9 @@ func (b *backendGRPCPluginServer) Setup(ctx context.Context, args *pb.SetupArgs)
 	var err error
 	id := singleImplementationID
 
+	b.instancesLock.Lock()
```

Comment:
Could this be moved down to line 114 to minimize locking time?

Reply:
Yep, I think it's safe to do so. Moved in https://github.com/hashicorp/vault/pull/17247/commits/6cf99cf949aac9f4227360641b3a453137345e1a.

Reply:
:+1: